code/main/edge/edgeRequestWorker.brs

```diff
@@ -132,16 +133,58 @@ function _adb_EdgeRequestWorker() as object
             return responseArray
         end function,
 
-        _processRequest: function(eventData as object, ecid as string, configId as string, requestId as string, path as string, edgeDomain = invalid as dynamic) as object
-            requestBody = m._createEdgeRequestBody(eventData, ecid)
+        _processRequest: function(eventData as object, ecid as string, datastreamId as string, requestId as string, path as string, meta as object, edgeDomain as dynamic) as object
+            ''' Append config overrides to meta if set in eventData.config
+            meta = m._appendConfigOverridesToMeta(meta, eventData.config, datastreamId)
 
-            url = _adb_buildEdgeRequestURL(configId, requestId, path, edgeDomain)
+            ''' Override the datastreamId if set in eventData.config
+            datastreamId = m._getDatastreamId(eventData.config, datastreamId)
+
+            ''' Remove config from eventData
+            eventData.Delete("config")
+
+            requestBody = m._createEdgeRequestBody(eventData, ecid, meta)
+
+            url = _adb_buildEdgeRequestURL(datastreamId, requestId, path, edgeDomain)
             _adb_logVerbose("EdgeRequestWorker::_processRequest() - Processing Request with url:(" + chr(10) + FormatJson(url) + chr(10) + ") with payload:(" + chr(10) + FormatJson(requestBody) + chr(10) + ")")
             networkResponse = _adb_serviceProvider().networkService.syncPostRequest(url, requestBody)
             return networkResponse
         end function
 
-        _createEdgeRequestBody: function(eventData as object, ecid as string) as object
+        _getDatastreamId: function(config as object, originalDatastreamId as String) as string
```

Comment:
nit: can we rename it to `_getOverriddenDatastreamId` ?

Reply:
So this method would either send overriddenDatastremId or the original if not present. That's why named it like this.